quarkus.ls.ext/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/extensions/ollama/OllamaItemMetadataProvider.java

```diff
@@ -0,0 +1,197 @@
+			hint.setName(QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY);
+			hint.setValues(Collections.emptyList());
+			projectInfo.getHints().add(hint);
+		}
+	}
+
+	@Override
+	public boolean isAvailable() {
+		return available;
+	}
+
+	@Override
+	public void update(PropertiesModel document) {
+		// Called when application.properties file is opened or updated (when user type
+		// something in the file).
+		boolean hasModelProperty = getProperty(document, QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY) != null;
+		if (hasModelProperty) {
+			// The application.properties declare the
+			// 'quarkus.langchain4j.ollama.chat-model.model-id' property,
+			// we need to collect available Ollama models by using Ollama Search Api
+			String searchUrl = getOllamaSearchUrl(document);
+			if (!Objects.equals(searchUrl, currentSearchUrl)
+					|| (System.currentTimeMillis() - lastDate) > REFRESH_OLLAMA_CACHE_TIMEOUT) {
+				// The current search url is different with the new, collect Ollama models.
+				ItemHint hint = projectInfo.getHint(QUARKUS_LANGCHAIN4J_OLLAMA_CHAT_MODEL_MODEL_ID_KEY);
+				// Update the available models values for the property
+				// 'quarkus.langchain4j.ollama.chat-model.model-id'
+				hint.setValues(collectOllamaModels(searchUrl));
+				currentSearchUrl = searchUrl;
+				lastDate = System.currentTimeMillis();
+			}
+		}
+	}
+
+	@Override
+	public List<ItemMetadata> getProperties() {
+		// This provider doesn't contribute to add/remove properties
+		return null;
+	}
+
+	private static List<ValueHint> collectOllamaModels(String searchUrl) {
+		try {
+			// Call Http search Url (ex : http://localhost:11434/api/tags)
+			HttpRequest request = HttpRequest.newBuilder() //
+					.uri(new URI(searchUrl)) //
+					.GET() //
+					.build();
+
+			HttpResponse<String> response = HttpClient.newBuilder() //
+					.followRedirects(HttpClient.Redirect.NORMAL) // follow redirects
+					.build() //
+					.send(request, BodyHandlers.ofString());
+
+			if (response.statusCode() != 200) {
+				return Collections.emptyList();
+			}
+
+			// ex :
+			/*
+			 * { models: [ { name: "qwen2:latest", model: "qwen2:latest", modified_at:
+			 * "2024-08-20T18:56:28.9319459+02:00", size: 4431400262, digest:
+			 * "e0d4e1163c585612b5419eee0a0d87fe0cac5fa6844001392e78d0cdf57c8138", details:
+			 * { parent_model: "", format: "gguf", family: "qwen2", families: [ "qwen2" ],
+			 * parameter_size: "7.6B", quantization_level: "Q4_0" } } ] }
+			 */
+			String result = response.body();
+			OllamaModelsResult r = new GsonBuilder() //
+					.create()//
+					.fromJson(new StringReader(result), OllamaModelsResult.class);
+			if (r != null && r.getModels() != null) {
+				return r.getModels() //
+						.stream() //
+						.map(m -> {
+							ValueHint model = new ValueHint();
+							String modelId = m.getName();
+							if (modelId.endsWith(":latest")) {
+								modelId = modelId.substring(0, modelId.length() - ":latest".length());

```

Comment:
modelId.substring(0, modelId.length() - 7); // with 7 == ":latest".length()
 
No need to recompute it every time

Reply:
fixed